### PR TITLE
STORY-18706 upgrade to gem best practices

### PR DESCRIPTION
**Pull request by ttstarck**

### CODE REVIEW

File: spec/elasticsearch_cluster_spec_helper.rb [hunk 1]
```diff
@@ -9,45 +9,37 @@ module ElasticsearchClusterSpecHelper
...
 
-    after(:all) do
-      if ELASTICSEARCH_TEST_PORT == 9250 || ELASTICSEARCH_TEST_LOCALHOST != '127.0.0.1'
-        Elasticsearch::Extensions::Test::Cluster.stop(CLUSTER_COMMANDS.merge(port: ELASTICSEARCH_TEST_PORT))
+      count = 0
+      until @elasticsearch_test_client.ping
+        count += 1
+        raise "Elasticsearch not ready after 60 tries" if count > 60
+        puts "Waiting for Elasticsearch to be ready... Attempt #{count}, sleeping 1 second"
+        sleep 1
```
Comment (ttstarck) [inline spec/elasticsearch_cluster_spec_helper.rb:42]: The Elasticsearch container can take 10-15 seconds to be ready by the time we hit this point

File: spec/elasticsearch_cluster_spec_helper.rb [hunk 2]
```diff
@@ -9,45 +9,37 @@ module ElasticsearchClusterSpecHelper
...
   def clear_and_create_index(index: ELASTICSEARCH_TEST_INDEX)
     if @elasticsearch_test_client&.indices&.exists?(index: index)
-      @elasticsearch_test_client.indices.delete(index: index)
-    end
-
-    unless @elasticsearch_test_client&.indices&.exists?(index: index)
-      @elasticsearch_test_client.indices.create(index: index)
+      @elasticsearch_test_client.indices.refresh(index: index)
+      @elasticsearch_test_client.delete_by_query(index: index, body: {
+        query: {
```
Comment (ttstarck) [inline spec/elasticsearch_cluster_spec_helper.rb:18]: This is faster for local development than deleting and recreating the entire index.

File: spec/elasticsearch_cluster_spec_helper.rb [hunk 3]
```diff
@@ -9,45 +9,37 @@ module ElasticsearchClusterSpecHelper
...
-  # WebMock.disable_net_connect!(allow_localhost: true) # allow local requests in setup
-  # WebMock.disable_net_connect! # disallow all requests in teardown
-
-  CLUSTER_COMMANDS =
-    if ENV["SEMAPHORE_CI_ELASTICSEARCH"].present?
-      { command: "/usr/share/elasticsearch/bin/elasticsearch", es_params: "-E path.conf=/etc/elasticsearch/" }
-    else
-      {}
-    end.freeze
+  ELASTICSEARCH_TEST_PORT      = ENV["ELASTICSEARCH_TEST_PORT"]&.to_i || 9200
```
Comment (ttstarck) [inline spec/elasticsearch_cluster_spec_helper.rb:12]: Port 9200 is used in the `docker-compose` file and the default port for our Elasticsearch container on the buildkite agent. There isn't any reason to have this default to 9250 anymore

**Review by dcaddell — COMMENTED** [review]
Changes seem reasonable to me, but I think we need to either make this gem private or move it off of buildkite?

**Comment by ttstarck** [comment]
@dcaddell I'm not going to worry too much about that. Its already public, and we're using buildkite. There aren't any secrets that we need to hide in the buildkite pipeline, so 🤷

**Review by dcaddell — APPROVED** [review]

**Pull request merged**